registry/config.go

```diff
@@ -89,45 +89,68 @@ func newServiceConfig(options ServiceOptions) *serviceConfig {
 		ServiceConfig: registrytypes.ServiceConfig{
 			InsecureRegistryCIDRs: make([]*registrytypes.NetIPNet, 0),
 			IndexConfigs:          make(map[string]*registrytypes.IndexInfo, 0),
-			// Hack: Bypass setting the mirrors to IndexConfigs since they are going away
-			// and Mirrors are only for the official registry anyways.
 		},
 		V2Only: options.V2Only,
 	}
 
-	config.LoadMirrors(options.Mirrors)
-	config.LoadInsecureRegistries(options.InsecureRegistries)
+	// Copy mirror information to IndexConfigs
+	err := config.LoadMirrors(options.Mirrors)
+	if err != nil {
+		panic(fmt.Sprintf("error loading mirrors: %s\n", err.Error()))
```

Comment:
Panic seemed reasonable to me, since mirrors are only read when the daemon starts up. The alternative seems to be failing silently, and ignoring invalid mirrors. What do you think is the best way to handle this?